File: 7,8 kyu/NextPerfectSquare.py
# https://www.codewars.com/kata/56269eb78ad2e4ced1000013/train
# Complete the findNextSquare method that finds the next integral perfect square after the one 
# passed as a parameter. Recall that an integral perfect square is an integer n such that sqrt(n) 
# is also an integer. If the parameter is itself not a perfect square then -1 should be returned. 
# You may assume the parameter is non-negative.

import math

def find_next_square(sq):
    # Return the next square if sq is a square, -1 otherwise
    rt = int(math.sqrt(sq))
    if sq == rt * rt:
        return (rt + 1) * (rt + 1)
    return -1

File: 7,8 kyu/NextPerfectSquare.java
// https://www.codewars.com/kata/56269eb78ad2e4ced1000013
// You might know some pretty large perfect squares. But what about the NEXT one?
// Complete the findNextSquare method that finds the next integral perfect square after 
// the one passed as a parameter. Recall that an integral perfect square is an integer n 
// such that sqrt(n) is also an integer. If the parameter is itself not a perfect square 
// then -1 should be returned. You may assume the parameter is non-negative.

public class NumberFun {
  public static long findNextSquare(long sq) {
    if(Math.sqrt(sq)==(int)Math.sqrt(sq)){
      long res = sq+1;
      boolean found = false;
      while(!found){
        if(Math.sqrt(res)==(int)Math.sqrt(res)){found=true; break;}
        res++;
      }
      return res;
    }
    else{return -1;} 
  }
}
